examples/Math.balls

```diff
@@ -0,0 +1,31 @@
+// Simple math example in BALLs
+
+extern _TERNARY() stack(3, 1)
+/** define _TERNARY macro in you huff output
+#define macro _TERNARY() = takes (3) returns (1) {
+    // input stack:     // [condition, true_value, false_value]
+    Jpm_TERNARY_is_zero jumpi
+
+    swap1               // [true_value, condition, false_value]
+    pop                 // [true_value]
```

Comment:
the inner pop is incorrect, the condition value gets consumed from the stack by the jumpi.

Generally for a ternary (assuming `condition` is a boolean value `1` / `0`) the XOR based ternary method should be cheaper:

```rust
fn TERNARY<z0>(true_value, condition, false_value) -> (result) {

    result = xor(
        false_value,
        mul(condition, xor(true_value, false_value))
    )
}
```

Reply:
Cool! i think i preffer a more traditional params syntax; `condition, true_value, false_value`